### ISSUE TITLE
Server error on ws.send

### ISSUE THREAD

**Issue by jasonford** [knowlearning/platform#23]
Looks like there is a condition where a send might be called on a CLOSING websocket, which results in an unhandled promise that kills the server.

Relevant Deno code: https://github.com/denoland/deno/blob/f698bc70e2f1d4cd58d17544258cf1b19726b66a/ext/websocket/01_websocket.js#L420

Seen error log:

ERROR 2024-05-06T08:32:46.156291673Z [0m[1m[31merror[0m: Uncaught (in promise) BadResource: Bad resource ID
ERROR 2024-05-06T08:32:46.156462574Z at [0m[1m[3mWebSocket.[[[eventLoop]]][0m ([0m[36mext:deno_websocket/01_websocket.js[0m:[0m[33m422[0m:[0m[33m19[0m)
ERROR 2024-05-06T08:32:46.156470834Z at [0m[1m[3meventLoopTick[0m ([0m[36mext:core/01_core.js[0m:[0m[33m65[0m:[0m[33m7[0m)

cleaned up error log:
```
ERROR 2024-05-06T08:32:46.156291673Z: Uncaught (in promise) BadResource: Bad resource ID
  at WebSocket.[eventLoop] (ext:deno_websocket/01_websocket.js:422:19)
  at eventLoopTick (ext:core/01_core.js:65:7)
```

Possible fix: make sure to ensure websocket is not CLOSING or CLOSED before sending.
Location for proposed fix: https://github.com/knowlearning/platform/blob/0bfe5f84dbc0f556a7039114937a0ee60a4aeac6/core/source/handle-http-request.js#L48

**Comment by jasonford** [comment]
Looks like there was a fix for a similar issue: https://github.com/denoland/deno/pull/21431/files

**Comment by jasonford** [comment]
That was a fix that was included in the versionof Deno we were running (1.40.4). Looks like there was an even more recent fix: https://github.com/denoland/deno/pull/21849 that fixes the issue we saw in production for https://github.com/pila-project/pila-platform/issues/126

Bumping the Deno version should fix
